diff --git a/Model/Import/Category.php b/Model/Import/Category.php
index 437cb59..f14c2fb 100644
--- a/Model/Import/Category.php
+++ b/Model/Import/Category.php
@@ -1063,8 +1063,11 @@ protected function getCategoryName($rowData)
      * @param string $string
      * @return array
      */
-    protected function explodeEscaped($delimiter = '/', $string)
+    protected function explodeEscaped($delimiter, $string)
     {
+        if (!$delimiter) {
+            $delimiter = '/';
+        }
         $exploded = explode($delimiter, $string);
         $fixed = [];
         for ($k = 0, $l = count($exploded); $k < $l; ++$k) {
